cli/tests/pcluster/utils.py

```diff
@@ -46,3 +49,33 @@ def get_resources(
             )
         )
     )
+
+
+def assert_lambdas_have_expected_vpc_config_and_managed_policy(generated_template, expected_vpc_config):
+    resources = generated_template.get("Resources")
+
+    for lambda_function in _get_lambda_functions(resources):
+        role = resources.get(_get_role_name(lambda_function))
+
+        if expected_vpc_config:
+            assert_that(_get_vpc_config(lambda_function)).is_equal_to(expected_vpc_config)
+            assert_that(_get_managed_policy_arns(role)).contains(LAMBDA_VPC_ACCESS_MANAGED_POLICY)
+        else:
+            assert_that(_get_vpc_config(lambda_function)).is_none()
+            assert_that(_get_managed_policy_arns(role)).does_not_contain(LAMBDA_VPC_ACCESS_MANAGED_POLICY)
+
+
+def _get_vpc_config(lambda_function):
+    return lambda_function.get("Properties").get("VpcConfig")
+
+
+def _get_role_name(lambda_function):
+    return lambda_function.get("Properties").get("Role").get("Fn::GetAtt")[0]
```

Comment:
Note: when you have multiple getters you can avoid KeyError or None exceptions in this way and fail according to the returned value.
```python
return lambda_function.get("Properties", {}).get("Role", {}).get("Fn::GetAtt")[0]
```

Reply:
Yes, however in this case I expect there to be there because these are the unit tests whose content I control (it would be a problem if these were empty).